controller/src/controller/models/domain/rsptag.py

```diff
@@ -49,8 +51,22 @@
 #
 # Note that this is matched top to bottom.  In particular, the release
 # candidate images must precede the release images since they would otherwise
-# parse as a release image with non-empty "rest".
+# parse as a release image with non-empty "rest", and anything with an rsp
+# version tag must precede the same type, but without that tag, or it will
+# match a non-empty "rest".
+#
 _TAG_REGEXES = [
+    # r23_0_0_rc1_rsp1.2.3_c0020.001_20210513
+    (
+        RSPImageType.CANDIDATE,
+        re.compile(_CANDIDATE + _RSP + _CYCLE + _REST + "$"),
+    ),
+    # r23_0_0_rc1_rsp1.2.3_c0020.001
+    (RSPImageType.CANDIDATE, re.compile(_CANDIDATE + _RSP + _CYCLE + "$")),
+    # r23_0_0_rsp1.2.3_rc1_20210513
+    (RSPImageType.CANDIDATE, re.compile(_CANDIDATE + _RSP + _REST + "$")),
```

Comment:
Comments for the first two version numbers are wrong, I think (the `rc` part is before the `rsp` part, surely).